src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -165,7 +165,24 @@ private[hyperspace] object IndexSummary {
       entry.derivedDataset.properties.columns.included,
       entry.numBuckets,
       entry.derivedDataset.properties.schemaString,
-      entry.content.root,
+      indexDirPath(entry),
       entry.state)
   }
+
+  /**
+   * This method extracts out the first index leaf directory path from a log entry. A leaf directory
```

Comment:
"This method extracts the most top-level (or top-most) index directory which has at least one file from a log entry"? A "leaf directory" definition here is a bit weird since it could have sub-directories and "first" can vary depending how you traverse the tree.

Reply:
thanks @imback82 , done